### ISSUE TITLE
有关这个Synthetic Bone Lengths

### ISSUE THREAD

**Issue by Duke-good** [hs1ang-hsu/BLAPose#3]
请问这个模型中生成Synthetic Bone Lengths是用于在训练骨骼长度的时候进行数据增强的吗？

**Comment by hs1ang-hsu** [comment]
Yes, the synthetic bone lengths are for the bone length prediction model.

**Comment by Duke-good** [comment]
Thank you for your reply. There is a question that I still don’t understand. When predicting bone length, is it trained with the value of the synthetic bone length as a label? Another point is, when predicting bone length, should we predict a bone length for each action and replace it, or should we use one bone length for all actions?

**Comment by hs1ang-hsu** [comment]
Yes, we take synthetic bone lengths as labels. For the second problem, we predict the bone lengths for all actions. We trained our model with the entire Human3.6M dataset, including all actions.

The bone length model's training process may be confusing. For general HPE methods, they use model-predicted 2D keypoints as input and 3D poses as labels. In our process, we first generate synthetic bone lengths to improve the variety of bone lengths in Human3.6M. Then we treat these bone lengths as labels. For the input, we should generate 2D keypoints since the bone lengths are changed. We adjust the corresponding 3D poses with synthetic bone lengths and reproject them to 2D keypoints.

**Comment by Duke-good** [comment]
Is the bone length of the human3.6 data set also used as a label? Or just use synthetic bone lengths as labels?

**Comment by Duke-good** [comment]
骨骼长度生成之后是一个什么维度的数据？

**Comment by Duke-good** [comment]
What dimension of data is generated after the bone length is generated?

**Comment by hs1ang-hsu** [comment]
The Human3.6M dataset uses 3D human poses as labels. The 3D human poses are obtained with MoCap system. Thus they are not synthetic.

For general usage of Human3.6M, a single data pair consists of 2D keypoints of size (frames, 17, 2) and 3D pose of size (1, 17, 3). In our case, a single set of synthetic bone lengths is of size (1, 16). A single data pair consists of 2D keypoints in (frames, 17, 2) and synthetic bone lengths (1, 16).

**Comment by Duke-good** [comment]
The dimension of the length_dict generated after bone length prediction is (236, 16). What does 236 mean?

**Comment by hs1ang-hsu** [comment]
Sorry, now I get what you mean. There are 236 videos in the test set of Human3.6M. We save as (236, 16) for convenience, but there are only two subjects in the test set. You may check that the first 120 rows are identical and the other 116 rows are also identical.

**Comment by Duke-good** [comment]
Oh oh, then I understand, thank you for your reply
